### ISSUE TITLE
9-slice sprite is behaving weirdly on UI when sprite size is smaller than texture size

### ISSUE THREAD

**Issue by Litttlefish** [bevyengine/bevy#11944]
### Discussed in https://github.com/bevyengine/bevy/discussions/11937

<div type='discussions-op-text'>

<sup>Originally posted by **Litttlefish** February 18, 2024</sup>
I'm using the 9-slice feature on 0.13 to rewrite a counter ui, but when using it, it give me a weird result that is not scaled down.
![image](https://github.com/bevyengine/bevy/assets/38809254/147ac426-74d2-4c10-a2f1-e4b178f9c79f)
If I remove ImageScaleMode component, then it will give me a sprite that is correctly resized.
![图片](https://github.com/bevyengine/bevy/assets/38809254/7a0cbcea-af29-4872-8578-af03c5382798)
Also I tried using it with SpriteBundle, seems working normally
![图片](https://github.com/bevyengine/bevy/assets/38809254/da1338c0-4c89-4295-9cf4-73d05caa1c8f)
Is it a bug, or this feature just don't support it?
</div>

**Comment by Litttlefish** [comment]
Further information:
image size is 240*120, while the ui node size should be 30px high and fits text length, `BorderRect` size is 48 px squared

**Comment by Litttlefish** [comment]
An addendum:
seems ButtonBundle works fine, only ImageBundle has this problem(source from bevy discord)
![image](https://github.com/bevyengine/bevy/assets/38809254/382a6a05-7231-45d2-bed6-488d123ee785)

**Comment by ManevilleF** [comment]
Could you provide some code to reproduce ? I made the slicing feature so I might be able to help and fix it

**Comment by Litttlefish** [comment]
> Could you provide some code to reproduce ? I made the slicing feature so I might be able to help and fix it

I think the 9-slicing example can tell by replacing `ButtonBundle` to `ImageBundle` or `AtlasImageBundle`.

 ![image](https://private-user-images.githubusercontent.com/38809254/305865073-382a6a05-7231-45d2-bed6-488d123ee785.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDg2MDgyNzgsIm5iZiI6MTcwODYwNzk3OCwicGF0aCI6Ii8zODgwOTI1NC8zMDU4NjUwNzMtMzgyYTZhMDUtNzIzMS00NWQyLWJlZDYtNDg4ZDEyM2VlNzg1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjIyVDEzMTkzOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFkZmVhYzRkNGQxNjZjZjA5MmYwN2Q3Y2YzZGI2MTkxYmE0MWE4YTlhNDYwZjRiMjliNDA2MmMxMmExOTU1MzUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.SSkd_AEbCPHh4oyY3jpgC64MhInJQ2RxYHrMcXVKD6M)

Because the upper one in this image also has this problem, but the actual button is larger than the image size, so I will also rename the issue.

**Comment by ManevilleF** [comment]
@Litttlefish Yup it doesn't work with `ImageBundle`, my bad this is due to a mistake in my implementation, I'll open a PR right away

**Comment by ManevilleF** [comment]
@Litttlefish Can you try with the fix ?

**Comment by Litttlefish** [comment]
> @Litttlefish Can you try with the fix ?

Tried, seems it works normally now.
![图片](https://github.com/bevyengine/bevy/assets/38809254/9eab656c-a839-4008-a8bd-a8de02c98b62)
